### ISSUE TITLE
Missing field "query-input"

### ISSUE THREAD

**Issue by ajoudeh-personal** [simialbi/yii2-schema-org#10]
Hi,

I use below code to generate WebSite schema

`
$siteLinks->potentialAction = new SearchAction(['target' => Yii::$app->urlManager->createAbsoluteUrl(['/site/search/', 'q' => '{search_term_string}']), 'query' => 'required name=search_term_string']);
`

i got this error from Google Search Test:

Missing field "query-input"

Any Advice?

Thanks

**Comment by ajoudeh-personal** [comment]
yes i see schema doc site, but when i fix it as workaround, its be valid in both site validation Google & Schema.

my work around is to pass `potentialAction` as json `["@type" => "SearchAction", 'target' => Yii::$app->urlManager->createAbsoluteUrl(['/site/search/']) . "?q={search_term_string}", 'query-input' => 'required name=search_term_string'];`

<img width="867" alt="Screenshot 2023-10-09 at 3 17 49 PM" src="https://github.com/simialbi/yii2-schema-org/assets/103322246/0ca46169-04d0-49b7-9a87-85a227a69bca">

**Comment by simialbi** [comment]
yeah but it's not even schema org property compliant (id-syntax instead of camelCase). Google 🤦🏻‍♂️...
Maybe user `DynamicModel` as base to be able to add properties 🤔
